src/Test.UnitTests.Sarif/Writers/BaseLoggerTests.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+using FluentAssertions;
+
+using Microsoft.CodeAnalysis.Sarif;
+
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Test.UnitTests.Sarif.Writers
+{
+    public class BaseLoggerTests
```

Comment:
>BaseLoggerTests [](start = 17, length = 15)

These tests are comprehensive enough. Please add a test for this user configuration:

--kind Open;Fail --level Error

Your code currently won't work for this combination (which should allow through any Open or Fail result (but only Fail messages of level Error)